src/serac/physics/materials/thermal_functional_material.hpp

```diff
@@ -88,6 +88,89 @@ class LinearIsotropicConductor {
   double conductivity_;
 };
 
+/// Linear isotropic thermal conduction material model
+class ParameterizedLinearIsotropicConductor {
+public:
+  /**
+   * @brief Construct a new Parameterized Linear Isotropic Conductor object
+   *
+   * @param density Density of the material (mass/volume)
+   * @param specific_heat_capacity Specific heat capacity of the material (energy / (mass * temp))
+   * @param conductivity Thermal conductivity of the material (power / (length * temp))
+   */
+  ParameterizedLinearIsotropicConductor(double density = 1.0, double specific_heat_capacity = 1.0,
+                                        double conductivity = 1.0)
+      : density_(density), specific_heat_capacity_(specific_heat_capacity), conductivity_(conductivity)
+  {
+    SLIC_ERROR_ROOT_IF(conductivity_ < 0.0,
+                       "Conductivity must be positive in the linear isotropic conductor material model.");
+
+    SLIC_ERROR_ROOT_IF(density_ < 0.0, "Density must be positive in the linear isotropic conductor material model.");
+
+    SLIC_ERROR_ROOT_IF(specific_heat_capacity_ < 0.0,
+                       "Specific heat capacity must be positive in the linear isotropic conductor material model.");
+  }
+
+  /**
+   * @brief Function defining the thermal flux (constitutive response)
+   *
+   * @tparam T1 type of the temperature (e.g. tensor or dual type)
+   * @tparam T2 type of the temperature gradient (e.g. tensor or dual type)
+   * @tparam T3 type of the parameter field (e.g. tensor or dual type)
+   * @param temperature_gradient Gradient of the temperature (du_dx)
+   * @param parameter The user-defined parameter used to calculate the thermal conductivity
+   * @return The thermal flux of the material model
+   */
+  template <typename T1, typename T2, typename T3>
+  SERAC_HOST_DEVICE auto operator()(const T1& /* temperature */, const T2& temperature_gradient,
+                                    const T3& parameter) const
```

Comment:
I imagine the intent is that `T1`, `T2` and `T3` are short for Type1, Type2, and Type3. I was briefly confused because I thought `T1` was chosen because it was the type of the temperature, and `T2` was the type of the temperature gradient.  Why it's the third argument `P1` because it's the first parameter?

Reply:
`T` is to `typename` template arguments as `i` is to loop indices: a not-actually-useful, but conventional choice for a name. See:
https://en.cppreference.com/w/cpp/container/vector
https://en.cppreference.com/w/cpp/container/array

Another common convention is to use `S`, `T`, `U`, `V` for a sequence of types .. in the same way nested loops often get `i`, `j`, `k`, `l`, ...

An alternate implementation could be
```cpp
  template <typename temperature_type, typename temperature_gradient_type, typename parameter_type>
  SERAC_HOST_DEVICE auto operator()(const temperature_type&  temperature,
                                    const temperature_gradient_type& temperature_gradient,
                                    const parameter_type& parameter) const { ... }
```
but that looks overly verbose to my eyes.

Reply:
For me, there is also a `T` is to temperature as `l` is to length or `m` is to mass going on here. It leads to a little cognitive dissonance. `T1` -> temperature, `T2` -> temperature gradient, `T3` -> parameter one :scream_cat:

Reply:
Unfortunately, the computer-science naming convention is conflicting with the physics naming convention here, so whichever one we choose will make someone angry! I think serac is sort of leaning more toward the computer science conventions, because they tend to be more self documenting (e.g. calling something `temperature` rather than `u` or `T`, "conductivity" rather than `k`, etc.), which I think is beneficial when you consider that a lot of the people that work on serac are not subject matter experts on the actual physics.

Reply:
I won't be angry.

Reply:
Yeah, since templates are very CS-y, I'm leaning toward the C++ conventions here.